wait-for-idle-runner/main.js

```diff
@@ -0,0 +1,27 @@
+const { Octokit } = require("@octokit/action");
+
+const octokit = new Octokit();
+
+const NUMBER_OF_ATTEMPTS = 3000;
+const TIME_BETWEEN_ATTEMPTS_SECONDS = 5000;
+
+async function main() {
+  for (var i=0; i < NUMBER_OF_ATTEMPTS; i++) {
+    const result = await octokit.request(`GET /repos/${process.env.REPOSITORY_NAME}/actions/runners`)
+
+    // Select the runner based on it's name
+    runner = result.data["runners"].filter(function(runner) {
+      return runner.name == process.env.RUNNER_NAME;
+    })[0]
```

Comment:
What happens if the runner name isn't found?

Reply:
I added a new check for this. The runner variable will be undefined, and in that case I am using an early return now.